internal/state/state.go

```
@@ -99,15 +119,32 @@ func Path(homeDir string) string {
 	if err != nil {
+		// Absent / IO / permission: keep the raw error unwrapped so callers can
+		// distinguish it (e.g. errors.Is(err, os.ErrNotExist)) and avoid resetting
+		// a file whose bytes were merely unreadable this once.
 		return InstallState{}, err
 	}
 	var s InstallState
 	if err := json.Unmarshal(data, &s); err != nil {
-		return InstallState{}, err
+		// The bytes are present but undecodable — the persisted data is already
+		// lost. Wrap the sentinel so callers can safely reset to a fresh state.
+		return InstallState{}, fmt.Errorf("%w: %v", ErrCorruptState, err)
```

golangci-lint (2.12.2)

[error] non-wrapping format verb for fmt.Errorf. Use `%w` to format errors

(errorlint)